hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeType.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch
+
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.BooleanMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.DateMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.DateTimeMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.IntegerMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.StringMatcher
+import uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api.TypeMatcher
+import java.time.LocalDate
+import java.time.LocalDateTime
+import kotlin.reflect.KClass
+
+typealias Attributes = Map<String, AttributeType>
+
+enum class AttributeType(private val kclass: KClass<*>, private val matcherKclass: KClass<out TypeMatcher>) {
+  STRING(String::class, StringMatcher::class),
+  INTEGER(Integer::class, IntegerMatcher::class),
+  BOOLEAN(Boolean::class, BooleanMatcher::class),
+  DATE(LocalDate::class, DateMatcher::class),
+  DATE_TIME(LocalDateTime::class, DateTimeMatcher::class),
+  ;
```

Comment:
Pulled the supported simple types into an enum with their related classes to allow making things more generic. Should be easier to add new types.